### ISSUE TITLE
Target temperature setting is off

### ISSUE THREAD

**Issue by sciurius** [albaintor/homeassistant_electrolux_status#18]
AEG BPE535E70M Surround Cook Oven.

Target temperature is a slider 0..100, step 1. Sliding to the far right gives a temperature of 100°C.

The actual temperature range depends on the chosen cooking program, but is generally 80..300, step 5.

**Comment by Imperial-Guard** [comment]
+1 

Mentioned this also in the discussion: 

Not sure if this would be possible but currently the number target temperature has a max of 100C.
See also the screendump below, I was wondering if we can make this more dynamic and depending on oventype to set this to max 230C (Steam-Bakeoven) or 300C (Pyrolatic-Bakeoven)

I assume when we iterate trough the programs and look for the max value we can set the max number here for.
Each program is returning his max temperature.

Let me know if you need some more info, thanks :)

**Comment by sciurius** [comment]
It is still fixed at 0..100. Using the slider I cannot get higher than 100°C.

![scrot20240207103244](https://github.com/albaintor/homeassistant_electrolux_status/assets/1118804/04135264-0376-48df-9cb5-720bf9246fb5)

**Comment by albaintor** [comment]
Ok this is probably because the appliance don't report the range and step for this field. I will try to add it in the catalog

**Comment by sciurius** [comment]
The range varies per program and is reported there (leaving out not relevant entries):
````
  "program": {
    "access": "readwrite",
    "type": "string",
    "values": {
      "BOTTOM": {
        "startTime": {
          "access": "readwrite",
          "default": "INVALID_OR_NOT_SET_TIME",
          "disabled": false,
          "max": 86340,
          "min": 0,
          "step": 60,
          "values": {
            "INVALID_OR_NOT_SET_TIME": {
              "disabled": true
            }
          }
        },
        "targetDuration": {
          "access": "readwrite",
          "disabled": false,
          "max": 86340,
          "min": 0,
          "step": 60
        },
        "targetTemperatureC": {
          "access": "readwrite",
          "default": 150.0,
          "disabled": false,
          "max": 300.0,
          "min": 80.0,
          "step": 5.0,
          "type": "temperature"
        }
      },
      "CONVENTIONAL_COOKING": {
        "startTime": {
          ...
        },
        "targetDuration": {
          ...
        },
        "targetTemperatureC": {
          "access": "readwrite",
          "default": 200.0,
          "disabled": false,
          "max": 300.0,
          "min": 30.0,
          "step": 5.0,
          "type": "temperature"
        }
      },
````

As a quick fix you can take the highest max and lowest min of all programs.

**Comment by albaintor** [comment]
Yeah it is too complicated at this state to implement dependencies between entities. The quick fix will do for now (in 1.0.6)
